logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/user/IsPasswordRequiredUseCaseTest.kt

```diff
@@ -0,0 +1,104 @@
+package com.wire.kalium.logic.feature.user
+
+import com.wire.kalium.logic.StorageFailure
+import com.wire.kalium.logic.data.session.SessionRepository
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.util.shouldFail
+import com.wire.kalium.logic.util.shouldSucceed
+import com.wire.kalium.persistence.model.SsoIdEntity
+import io.mockative.Mock
+import io.mockative.any
+import io.mockative.given
+import io.mockative.mock
+import io.mockative.once
+import io.mockative.verify
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import okio.IOException
+import kotlin.test.Test
+import kotlin.test.assertFalse
+import kotlin.test.assertIs
+import kotlin.test.assertTrue
+
+@ExperimentalCoroutinesApi
+class IsPasswordRequiredUseCaseTest {
+
+    @Test
+    fun givenUSerHasNoSsoId_thenReturnTrue() = runTest {
+        val (arrangement, isPasswordRequired) = Arrangement()
+            .withSelfSsoId(Either.Right(null))
+            .arrange()
+
+        isPasswordRequired.eitherInvoke().shouldSucceed {
+            assertTrue(it)
+        }
+
+        verify(arrangement.sessionRepository)
+            .invocation { arrangement.sessionRepository.ssoId(arrangement.selfUserId) }
```

Comment:
Ideally, the `selfUserId` should be provided when initialising the arrangement, `withInitialUserId(currentUserId)`, which will update the value on each test init. That way we don't have to "guess" what user is being used under the hood to invoke the use case and we increase test readability.